packages/api/src/controllers/room.ts

```diff
@@ -17,18 +21,119 @@ app.post("/", authorizer({}), async (req, res) => {
     req.config.livekitApiKey,
     req.config.livekitSecret
   );
-  const opts = {
+  const room = await svc.createRoom({
     name: id,
     // timeout in seconds
-    emptyTimeout: 30 * 60,
-    maxParticipants: 2,
-  };
-  const room = await svc.createRoom(opts);
+    emptyTimeout: 15 * 60,
+    maxParticipants: 10,
+    metadata: `{"userId": "${req.user.id}"}`,
+  });
   console.log("livekit room created", room);
+
+  await db.room.create({
+    id: id,
+    userId: req.user.id,
+    participants: {},
+    events: [],
+  });
+
   res.status(201);
   res.json({
     id: id,
   });
 });
 
+app.get("/:roomId", authorizer({}), async (req, res) => {
+  const room = await db.room.get(req.params.roomId);
+
+  if (!req.user.admin && req.user.id !== room.userId) {
+    throw new ForbiddenError(`users may only view their own rooms`);
+  }
+
+  res.status(200).json(toExternalRoom(room, req.user.admin));
+});
+
+function toExternalRoom(room: Room, isAdmin = false) {
+  if (isAdmin) {
+    return room;
+  }
+  room.userId = undefined;
+  for (const key in room.participants) {
+    room.participants[key].tracksPublished = undefined;
+  }
+  room.events = undefined;
+  return room;
+}
+
+// Implement a webhook handler to receive webhooks from Livekit to update our state with room and participant details.
+app.use("/webhook", express.raw({ type: "*/*" }));
+app.post("/webhook", async (req, res) => {
+  const receiver = new WebhookReceiver(
+    req.config.livekitApiKey,
+    req.config.livekitSecret
+  );
+  console.log("livekit webhook auth", req.get("Authorization"));
+
+  // event is a WebhookEvent object
+  const event = receiver.receive(req.body, req.get("Authorization"));
+
+  let room: WithID<Room>;
+  switch (event.event) {
+    case "track_published":
+    case "participant_joined":
+    case "participant_left":
+      const roomId = event.room.name;
+      const svc = new RoomServiceClient(
+        req.config.livekitHost,
+        req.config.livekitApiKey,
+        req.config.livekitSecret
+      );
+      const participants = await svc.listParticipants(roomId);
+
+      room = await db.room.get(roomId);
```

Comment:
Maybe double check this room exists and return a 4xx error if it doesn't (or anything that LiveKit won't retry)